tests/v2/validation/fleet/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,210 @@
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/v2/actions/fleet"
+	projectsapi "github.com/rancher/rancher/tests/v2/actions/projects"
+	"github.com/rancher/rancher/tests/v2/actions/ssh"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	extensionClusters "github.com/rancher/shepherd/extensions/clusters"
+	extensionsfleet "github.com/rancher/shepherd/extensions/fleet"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/nodes"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type UpgradeTestSuite struct {
+	suite.Suite
+	client          *rancher.Client
+	session         *session.Session
+	cluster         *management.Cluster
+	clusterName     string
+	sshNode         *nodes.Node
+	fleetSecretName string
+}
+
+func (u *UpgradeTestSuite) TearDownSuite() {
+	u.session.Cleanup()
+}
+
+func (u *UpgradeTestSuite) SetupSuite() {
+	u.session = session.NewSession()
+
+	client, err := rancher.NewClient("", u.session)
+	require.NoError(u.T(), err)
+
+	u.client = client
+
+	log.Info("Getting cluster name from the config file and append cluster details in connection")
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(u.T(), clusterName, "Cluster name to install should be set")
+
+	clusterID, err := extensionClusters.GetClusterIDByName(u.client, clusterName)
+	require.NoError(u.T(), err, "Error getting cluster ID")
+
+	u.cluster, err = u.client.Management.Cluster.ByID(clusterID)
+	require.NoError(u.T(), err)
+
+	provisioningClusterID, err := extensionClusters.GetV1ProvisioningClusterByName(client, clusterName)
+	require.NoError(u.T(), err)
+
+	cluster, err := client.Steve.SteveType(extensionClusters.ProvisioningSteveResourceType).ByID(provisioningClusterID)
+	require.NoError(u.T(), err)
+
+	newCluster := &provv1.Cluster{}
+	err = steveV1.ConvertToK8sType(cluster, newCluster)
+	require.NoError(u.T(), err)
+
+	u.clusterName = client.RancherConfig.ClusterName
+	if !strings.Contains(newCluster.Spec.KubernetesVersion, "k3s") && !strings.Contains(newCluster.Spec.KubernetesVersion, "rke2") {
+		u.clusterName = u.cluster.ID
+	}
+
+	steveClient, err := u.client.Steve.ProxyDownstream(u.cluster.ID)
+	require.NoError(u.T(), err)
+
+	u.sshNode, err = ssh.CreateSSH(u.client, steveClient, u.cluster.Name, u.cluster.ID)
+	require.NoError(u.T(), err)
+
+	u.fleetSecretName, err = createFleetSSHSecret(u.client, string(u.sshNode.SSHKey))
+	require.NoError(u.T(), err)
+}
+
+func (u *UpgradeTestSuite) TestNewCommitFleetRepo() {
+	u.session = session.NewSession()
+
+	log.Info("Creating new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(u.client, u.cluster.ID)
```

Comment:
if project is not needed why create the project? instead can we not use the same project and create new namespaces in it?